### PR TITLE
Fix import overriding private setting

### LINKED ISSUE #10788: Imports override privacy mode setting
I'm certain this has come up before in the past, but not sure if tracked anywhere useful. Currently, if you import a JSON content file into Ghost, it also imports and overrides your currently active "private site mode" setting.

So what ends up happening is you're working on a new site, privately, and you import content from an old/existing site - and now suddenly your private site is public, without warning. There's actually no way of knowing anything happened at all unless you happen to notice that the setting has changed, or your site starts being indexed in Google.

This is a pretty bad bug.

IMO the site privacy setting should never be changed by any import - it should stay exactly as it has been manually set by the user.

### PR DESCRIPTION
**Pull request by allouis**
closes #10788 

- This removes the `is_private` setting from imports under *all* conditions.
- This outputs a "problem" when the import data's `is_private` settings differs from the current databases `is_private` setting.

### CODE REVIEW

**Comment by gargol** [comment]
yo @allouis what's the status of this PR? Looks like the tests need fixing and it's gonna be ready to :rocket:

**Comment by allouis** [comment]
Yo! Tests need fixing, but actually needs a lot more work, Zimo and John wanted some design changes, like moving this error to the top of the errors list, which will require a refactor. I can pick up some more dev work next week, but I can't see this being merged until Zimo is back - I could be wrong about that tho

**Comment by gargol** [comment]
Hm. Can the task be split into couple parts maybe? 1st would get this change in, so people don't accidentally override the private setting and have a follow up with design improvement? But maybe it's a no-go with this warning being lost in the clutter of other ones?

**Comment by allouis** [comment]
> But maybe it's a no-go with this warning being lost in the clutter of other ones?

I think that was John's main issue with it - let's see what he thinks, @JohnONolan any thoughts on the above discussion?

**Comment by JohnONolan** [comment]
Agree with @gargol - we can polish up the warning design later, be good to get the rest in now 👍

**Comment by allouis** [comment]
Will finish this Monday for a merge :relaxed:

**Comment by gargol** [comment]
Just making sure the behavior implemented is expected one. There are 4 scenarios with different outcomes at the moment:
1. Import of `is_private: false` into the site with `is_private: false` - no change.
2. Import of `is_private: false` into the site with `is_private: true` - we keep the site private as intended BUT! if the import contains a `password` field from settings it gets overridden (this doesn't seem right). 
3. Import of `is_private: true` into the site with `is_private: true` - same behavior as pt.2
4. Import of `is_private: true` into the site with `is_private: false` - the site stays public + show the warning message

Think pts 2 and 3 should not be resetting the password (needs to be filtered out along with `is_private` flag when importing)

File: core/server/data/importer/importers/data/settings.js [hunk 1]
```diff
@@ -80,6 +80,22 @@ class SettingsImporter extends BaseImporter {
...
 
+        const newIsPrivate = _.find(this.dataToImport, {key: 'is_private'});
+        const oldIsPrivate = _.find(this.existingData, {key: 'is_private'});
+
+        this.dataToImport = _.filter(this.dataToImport, (data) => {
+            return data.key !== 'is_private';
+        });
+
+        // Only show warning if we are importing a private site into a non-private site.
+        if (oldIsPrivate === false && newIsPrivate === true) {
```
Comment (gargol) [inline core/server/data/importer/importers/data/settings.js]:
```suggestion
        if ((oldIsPrivate && newIsPrivate) && (!!oldIsPrivate.value === false && !!newIsPrivate.value === true)) {

```

The check should take into account that these are models being compared `.value` bit, and that they might not be present in the imported file and the `value` might also be a string in the incoming data :grimacing:

**Review by naz — CHANGES_REQUESTED** [review]

File: core/server/data/importer/importers/data/settings.js (hunk 1 above)
Comment (gargol) [reply core/server/data/importer/importers/data/settings.js]: P.S. might be worth to get the new behavior covered a little with unit tests if not too much extra work

**Comment by allouis** [comment]
@gargol This is good to go IMO :tada:

**Review by naz — APPROVED** [review]

**Comment by gargol** [comment]
@allouis :ship: it

**Pull request merged**